eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -470,56 +479,61 @@ fixupFees tx = do
       mkBasicTxOut @era
         (mkAddr (kpSpending, kpStaking))
         (inject remainingCoin)
-  impRootTxCoinL .= remainingCoin
   let
-    balancedTx =
-      setMinFeeTx pp tx
+    txNoWits =
+      tx
         & bodyTxL . inputsTxBodyL %~ Set.insert (TxIn impRootTxId $ TxIx 0)
         & bodyTxL . outputsTxBodyL %~ (remainingTxOut :<|)
-    txId = TxId . hashAnnotated $ balancedTx ^. bodyTxL
-  impRootTxIdL .= txId
-  wits <- mkTxWits (balancedTx ^. bodyTxL)
-  pure $ balancedTx & witsTxL .~ wits
+    nativeScriptKeyWitsCount = 0 -- TODO figure out wits for native scripts
+    outsWithoutRemaining = tx ^. bodyTxL . outputsTxBodyL
```

Comment:
I don't understand why this is called `outsWithoutRemaining`,  it seems to be set to all the outs via `outputsTxBodyL` ?

Reply:
I think the idea is that the remaining coin gets added to this later so it's not the final sequence of outputs.

Reply:
Yes, "remaining" is in the sense of "remainder of the money" or "change given back to you"

That was the naming Joosep adopted, so I just went with it as well

Reply:
Makes sense, thank you both :+1: